scripts/prog8.rb

```diff
@@ -0,0 +1,20 @@
+class Array
+    def power(x)
+        arr=[]
+        self.each do |value , temp=x|
+            final_val=1
+            while temp!=0
+                temp-=1
+                final_val*=value
+            end
```

Comment:
It's always good to know the basic implementation, but you can use `**` operator present in ruby. It does the same thing.